clouddq/classes/dq_rule_binding.py

```diff
@@ -334,11 +336,20 @@ def resolve_all_configs_to_dict(
             row_filter_config = self.resolve_row_filter_config(configs_cache)
             # resolve reference columns config
             if self.reference_columns_id:
+                include_all_reference_columns = False
                 include_reference_columns = self.resolve_reference_columns_config(
                     configs_cache
                 ).include_reference_columns
+                if include_reference_columns[0] == "*":
+                    column_names = bigquery_client.get_table_columns(
+                        table=table_entity.get_table_name(),
+                        project_id=table_entity.instance_name,
+                    )
```

Comment:
what kind of errors does `bigquery_client.get_table_columns()` throw and where are they being handled? e.g. what happens when `table_entity` doesn't exists, when connection fails, or when the current user does have sufficient authentication to connect? Are we assuming that if the code reached this point we shouldn't get those errors?

Reply:
if any of the above errors happen, will the expected behaviour be for the client itself to throw and error and exit, i.e. we don't need to do anything else here?

Reply:
yes the exception handling is done in the bigquery_client if the table entity doesn't exists or unable to access table entity.